File: ServoLink/ServoCommander/Data/HexConfig.cs
namespace ServoCommander.Data
{
    public class HexConfig
    {
        public const int LegsCount = 6;
        public const double CoxaMin = -65;
        public const double CoxaMax = 65;
        public const double CoxaLength = 29;

        public const double FemurMin = -105;
        public const double FemurMax = 75;
        public const double FemurLength = 57;

        public const double TibiaMin = -53;
        public const double TibiaMax = 90;
        public const double TibiaLength = 141;

        public static short[] OffsetX = { -54, -108, -54, 54, 108, 54 }; //RR RM RF LF LM LR
        public static short[] OffsetZ = { 93, 0, -93, 93, 0, -93 }; //RR RM RF LF LM LR

        public static double[] CoxaDefaultAngle = { -59.7, 0, 59.7, -59.7, 0, 59.7 }; //RR RM RF LF LM LR

        public static double[] DefaultLegsPosX = { 56, 111, 56, 56, 111, 56 }; //RR RM RF LF LM LR
        public static double[] DefaultLegsPosY = { 65, 65, 65, 65, 65, 65 }; //RR RM RF LF LM LR
        public static double[] DefaultLegsPosZ = { 96, 0, -96, 96, 0, -96 }; //RR RM RF LF LM LR

        public static bool[] CoxaAngleInv = { true, true, true, false, false, false }; //RR RM RF LF LM LR
        public static bool[] FemurAngleInv = { true, true, true, false, false, false }; //RR RM RF LF LM LR
        public static bool[] TibiaAngleInv = { true, true, true, false, false, false }; //RR RM RF LF LM LR LF LM LR

        public const double MaxBodyHeight = 100;
        public const double BodyStandUpOffset = 45;
        public const double LegLiftHeight = 55;
        public const double LegLiftDoubleHeight = 80;
        public const double GPlimit = 2;
        public const double TravelDeadZone = 4;
    }
}

File: sc32v3_esp32/lib/platform/HexConfig.cpp
#include "HexConfig.h"

int HexConfig::LegsCount = 6;
double HexConfig::MaxBodyHeight = 90;
double HexConfig::BodyInitY  = 65;

double HexConfig::CoxaMin = -65;
double HexConfig::CoxaMax = 65;
double HexConfig::CoxaLength = 29;
double HexConfig::FemurMin = -105;
double HexConfig::FemurMax = 75;
double HexConfig::FemurLength = 57;
double HexConfig::TibiaMin = -53;
double HexConfig::TibiaMax = 90;
double HexConfig::TibiaLength = 141;
short HexConfig::OffsetX[] = { -69, -138, -69, 69, 138, 69 }; //RR RM RF LF LM LR
short HexConfig::OffsetZ[] = { 119, 0, -119, 119, 0, -119 }; //RR RM RF LF LM LR
double HexConfig::CoxaDefaultAngle[] = { -60, 0, 60, -60, 0, 60 }; //RR RM RF LF LM LR

double HexConfig::DefaultLegsPosX[] = { 56, 111, 56, 56, 111, 56 }; //RR RM RF LF LM LR
double HexConfig::DefaultLegsPosY[] = { HexConfig::BodyInitY, HexConfig::BodyInitY, HexConfig::BodyInitY, HexConfig::BodyInitY, HexConfig::BodyInitY, HexConfig::BodyInitY }; //RR RM RF LF LM LR
double HexConfig::DefaultLegsPosZ[] = { 96, 0, -96, 96, 0, -96 }; //RR RM RF LF LM LR

int HexConfig::HexIntXZCount = 3;
double HexConfig::HexIntXZ[] = { 111, 99, 86 };
double HexConfig::HexMaxBodyY[] = { 20, 50, HexConfig::MaxBodyHeight };

bool HexConfig::CoxaAngleInv[] = { true, true, true, false, false, false }; //RR RM RF LF LM LR
bool HexConfig::FemurAngleInv[] = { true, true, true, false, false, false }; //RR RM RF LF LM LR
bool HexConfig::TibiaAngleInv[] = { true, true, true, false, false, false }; //RR RM RF LF LM LR LF LM LR

double HexConfig::BodyStandUpOffset = HexConfig::BodyInitY;
double HexConfig::LegLiftHeight = 40;
double HexConfig::LegLiftDoubleHeight = HexConfig::BodyInitY;
double HexConfig::GPlimit = 2;
double HexConfig::TravelDeadZone = 4;

uint16_t HexConfig::WalkingDelay =  200;
uint16_t HexConfig::BalancingDelay = 100;
uint16_t HexConfig::SingleLegControlDelay = 25;

int HexConfig::GaitsCount = 6;